src/OpenTelemetry.Exporter.Jaeger/Implimentation/InMemoryTransport.cs

```diff
@@ -0,0 +1,101 @@
+// <copyright file="InMemoryTransport.cs" company="OpenTelemetry Authors">
+// Copyright 2018, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+namespace OpenTelemetry.Exporter.Jaeger.Implimentation
+{
+    using System;
+    using System.IO;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Thrift.Transports;
+
+    internal class InMemoryTransport : TClientTransport
+    {
+        private readonly MemoryStream byteStream;
+        private bool isDisposed;
+
+        public InMemoryTransport()
+        {
+            this.byteStream = new MemoryStream();
+        }
+
+        public override bool IsOpen => true;
+
+        public override async Task OpenAsync(CancellationToken cancellationToken)
+        {
+            if (cancellationToken.IsCancellationRequested)
+            {
+                await Task.FromCanceled(cancellationToken);
+            }
+        }
+
+        public override void Close()
+        {
+            // do nothing
+        }
+
+        public override async Task<int> ReadAsync(
+            byte[] buffer,
+            int offset,
+            int length,
+            CancellationToken cancellationToken)
+        {
+            return await this.byteStream.ReadAsync(buffer, offset, length, cancellationToken);
```

Comment:
Personally I'd do: 
```suggestion
            => this.byteStream.ReadAsync(buffer, offset, length, cancellationToken);
```

Braces or not, method doesn't need to be `async`.
Same for all others in this file. If have no error handling, don't need change execution context or work with the return or continue after the async operations continues there's no need to make the method async.

Reply:
@bruno-garcia thanks for the "uninvited" input. It's literally been 5 years since I touched .NET, and I'm still catching up on the new additions to C#. This is great stuff. Thank you.